raiden/network/transport/matrix/utils.py

```diff
@@ -434,6 +434,11 @@ def login_or_register(
                 _exception=ex,
             )
         else:
+            # Login suceeded. Sync with the server to fetch the inventory rooms
+            # through the invites, however at this point the messages
```

Comment:
Humm, I'm missing something then.

Does this queries the rooms the user has joined previously? And that is done because we don't save the private rooms in the database so we have to sync them again after login?

Reply:
Yes, the server keeps track of the user's joined rooms.
It's not 'because' we don't locally store them, it's just a core feature of matrix. 
The idea is, that you will get all the same rooms if you log in from different devices (which isn't a use case we have in raiden).